src/utils/configuration.ts

```diff
@@ -0,0 +1,49 @@
+import * as dotenv from 'dotenv';
+
+export class Configurations {
+  private static instance: Configurations;
+
+  private constructor() {
+    dotenv.config();
+  }
+
+  public static getInstance(): Configurations {
+    if (!Configurations.instance) {
+      Configurations.instance = new Configurations();
+    }
+
+    return Configurations.instance;
+  }
+
+  get branch(): string {
+    return process.env.BRANCH || 'local';
+  }
+
+  get awsRegion(): string {
+    return process.env.AWS_PROVIDER_REGION || 'local';
+  }
+
+  get dynamoTableName(): string {
+    return `cvs-${this.branch}-trailer-registration`;
+  }
+
+  get dynamoParams(): unknown {
+    const dbParams = {
+      region: this.awsRegion,
+      endpoint: process.env.DYNAMO_ENDPOINT || 'http://localhost:8020',
```

Comment:
if we don't allow the port to be passed I think we can use the endpoint and remove it from the env as this environmental variable won't be needed. I think it is best if we pass the base url and port separately so we allow more granular configuration, for example some services depending on where they are deployed may share the same base url but need to have different port exposed for networking

Reply:
I have modified it but the config file for remote would end up quite different from the one in local as only one parameter i.e. `convertEmptyValue` is used in AWS.